module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -82,14 +85,37 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $localDirs = defined('LOCAL_OVERRIDE_DIR')
+        $localDirs = [];
+        $currentDir = defined('LOCAL_OVERRIDE_DIR')
             && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
-                ? [
-                    [
-                        'directory' => LOCAL_OVERRIDE_DIR,
-                        'defaultConfigSubdir' => $this->defaultLocalConfigSubdir,
-                    ],
-                ] : [];
+            ? LOCAL_OVERRIDE_DIR : '';
+        while (!empty($currentDir)) {
+            $systemConfigFile = $currentDir . '/DirLocations.ini';
+            $systemConfig = new Config(
+                file_exists($systemConfigFile)
+                    ? $this->getIniReader()->fromFile($systemConfigFile)
+                    : []
+            );
+            array_unshift(
+                $localDirs,
+                [
+                    'directory' => $currentDir,
```

Comment:
Should we check if $currentDir exists and is a directory before adding it to the stack? Maybe we should throw an exception if configuration points at something that doesn't exist or can't be used as a directory. Also, maybe it's worth throwing a `realpath` in somewhere to make the paths more readable if we do have messages about them.

Reply:
Using realpath to make the paths more readable. It can also be used directly to check if the paths exists. Now, an \Exception is thrown if a path does not exist. If you know a more specific exception type, I would change this. But right now, I could not think of a fitting one.

Throwing an Exception at this point results in an white screen and it's not directly obvious where the error comes from. Maybe a warning will be sufficient, since a non existing path will not directly result in errors?

Reply:
Yes, I think a warning would be appropriate in this situation if the alternative is a difficult-to-troubleshoot "white screen of death."

Reply:
I changed the exception to a warning.